### ISSUE TITLE
`Any` apparently not considered part of `Union[Any, builtins.None]`

### ISSUE THREAD

**Issue by OddBloke** [python/mypy#3472]
When running `mypy --strict` against the following code:

```py
from typing import Any, Optional

class Test(object):

    def __init__(self) -> None:
        self.attr = "foo"  # type: Any

    def foo(self, do_it: bool) -> Optional[Any]:
        if do_it:
            return self.attr
        else:
            return None
```

I see the following warning:

```
testcase.py:11: warning: Returning Any from function with declared return type "Union[Any, builtins.None]"
```

This seems obviously incorrect. At the very least, it's a confusing error message.

Bisecting suggests that this behaviour was introduced in 4141d2039c709f3a8e591fa35a560e54c4e203d2.

**Comment by gvanrossum** [comment]
That's @JukkaL's PR #3086.

**Comment by ilevkivskyi** [comment]
I think the problem is not with Jukka's PR but in the logic of ``--warn-return-any``. It should not emit warning if ``Any`` is a proper subtype of the declared return type. I have an idea for a very simple fix, if it will work, then I will make a PR shortly.

**Comment by ilevkivskyi** [comment]
OK, here is the PR https://github.com/python/mypy/pull/3473

**Comment by OddBloke** [comment]
Thanks all! When might we expect to see this in a release?

**Comment by JukkaL** [comment]
We typically have a release every 2 months or so, so the next release could be June/Julyish.
